internal/envoy/listener.go

```diff
@@ -65,47 +70,56 @@ func Listener(name, address string, port int, lf []listener.ListenerFilter, filt
 // HTTPConnectionManager creates a new HTTP Connection Manager filter
 // for the supplied route and access log.
 func HTTPConnectionManager(routename, accessLogPath string) listener.Filter {
+	duration := func(d time.Duration) *time.Duration {
+		return &d
+	}
 	return listener.Filter{
 		Name: util.HTTPConnectionManager,
-		ConfigType: &listener.Filter_Config{
-			Config: &types.Struct{
-				Fields: map[string]*types.Value{
-					"stat_prefix": sv(routename),
-					"rds": st(map[string]*types.Value{
-						"route_config_name": sv(routename),
-						"config_source": st(map[string]*types.Value{
-							"api_config_source": st(map[string]*types.Value{
-								"api_type": sv("GRPC"),
-								"grpc_services": lv(
-									st(map[string]*types.Value{
-										"envoy_grpc": st(map[string]*types.Value{
-											"cluster_name": sv("contour"),
-										}),
-									}),
-								),
-							}),
-						}),
-					}),
-					"http_filters": lv(
-						st(map[string]*types.Value{
-							"name": sv(util.Gzip),
-						}),
-						st(map[string]*types.Value{
-							"name": sv(util.GRPCWeb),
-						}),
-						st(map[string]*types.Value{
-							"name": sv(util.Router),
-						}),
-					),
-					"http_protocol_options": st(map[string]*types.Value{
-						"accept_http_10": {Kind: &types.Value_BoolValue{BoolValue: true}},
-					}),
-					"access_log":         accesslog(accessLogPath),
-					"use_remote_address": {Kind: &types.Value_BoolValue{BoolValue: true}}, // TODO(jbeda) should this ever be false?
-					"normalize_path":     {Kind: &types.Value_BoolValue{BoolValue: true}},
-					"idle_timeout":       sv("60s"),
+		ConfigType: &listener.Filter_TypedConfig{
+			TypedConfig: any(&http.HttpConnectionManager{
+				StatPrefix: routename,
+				RouteSpecifier: &http.HttpConnectionManager_Rds{
+					Rds: &http.Rds{
+						RouteConfigName: routename,
+						ConfigSource: core.ConfigSource{
+							ConfigSourceSpecifier: &core.ConfigSource_ApiConfigSource{
+								ApiConfigSource: &core.ApiConfigSource{
+									ApiType: core.ApiConfigSource_GRPC,
+									GrpcServices: []*core.GrpcService{{
+										TargetSpecifier: &core.GrpcService_EnvoyGrpc_{
+											EnvoyGrpc: &core.GrpcService_EnvoyGrpc{
+												ClusterName: "contour",
+											},
+										},
+									}},
+								},
+							},
+						},
+					},
 				},
-			},
+				HttpFilters: []*http.HttpFilter{{
+					Name: util.Gzip,
+				}, {
+					Name: util.GRPCWeb,
+				}, {
+					Name: util.Router,
+				}},
+				HttpProtocolOptions: &core.Http1ProtocolOptions{
```

Comment:
@stevesloka this comes from #537 which (sort of) enabled HTTP/1.0 support. I've got another PR stacked behind this one, so I'll add a comment inline to reference #537 in the next PR.